doit/cmd_clean.py

```diff
@@ -29,32 +29,46 @@
     'help': 'clean all task',
     }
 
+opt_clean_hard = {
+    'name': 'hard',
+    'long': 'hard',
+    'type': bool,
+    'default': False,
+    'help': 'also forget tasks after cleaning',
+    }
 
 class Clean(DoitCmdBase):
     doc_purpose = "clean action / remove targets"
     doc_usage = "[TASK ...]"
     doc_description = ("If no task is specified clean default tasks and "
                        "set --clean-dep automatically.")
 
-    cmd_options = (opt_clean_cleandep, opt_clean_cleanall, opt_clean_dryrun)
+    cmd_options = (opt_clean_cleandep, opt_clean_cleanall, opt_clean_dryrun, opt_clean_hard)
 
 
-    def clean_tasks(self, tasks, dryrun):
+    def clean_tasks(self, tasks, dryrun, hard):
         """ensure task clean-action is executed only once"""
         cleaned = set()
+        forget_tasks = hard and not dryrun
         for task in tasks:
             if task.name not in cleaned:
                 cleaned.add(task.name)
                 task.clean(self.outstream, dryrun)
+                if forget_tasks:
+                    self.dep_manager.remove(task.name)
 
+        if forget_tasks:
+            self.dep_manager.close()
 
-    def _execute(self, dryrun, cleandep, cleanall, pos_args=None):
+    def _execute(self, dryrun, cleandep, cleanall, hard, pos_args=None, clean_hard=False):
```

Comment:
You are not supposed to add 2 parameters. Why 2?

Reply:
The param with `name` as` hard` should be able to use it from both command-line and config. But as I suggested change it to `forget`. A `clean_XXX` param on `_execute` is not necessary.